src/NodeVisitor/EscaperNodeVisitor.php

```diff
@@ -82,6 +90,41 @@ protected function doLeaveNode(Node $node, Environment $env)
         return $node;
     }
 
+    private function unwrapConditional(ConditionalExpression $expression, Environment $env)
+    {
+        // convert "echo a ? b : c" to "a ? echo b : echo c" recursively
+        $expr2 = $expression->getNode('expr2');
+        if ($expr2 instanceof ConditionalExpression) {
+            $expr2 = $this->unwrapConditional($expr2, $env);
+        } else {
+            $expr2 = $this->escapeInlinePrintNode(new InlinePrint($expr2, $expr2->getTemplateLine(), $expr2->getNodeTag()), $env, $this->needEscaping($env));
+        }
+        $expr3 = $expression->getNode('expr3');
+        if ($expr3 instanceof ConditionalExpression) {
+            $expr3 = $this->unwrapConditional($expr3, $env);
+        } else {
+            $expr3 = $this->escapeInlinePrintNode(new InlinePrint($expr3, $expr3->getTemplateLine(), $expr3->getNodeTag()), $env, $this->needEscaping($env));
+        }
+
+        return new ConditionalExpression($expression->getNode('expr1'), $expr2, $expr3, $expression->getTemplateLine());
+    }
+
+    private function escapeInlinePrintNode(InlinePrint $node, Environment $env, $type)
+    {
+        $expression = $node->getNode('node');
+
+        if ($this->isSafeFor($type, $expression, $env)) {
+            return $node;
+        }
+
+        $class = \get_class($node);
+
+        return new $class(
```

Comment:
why making this dynamic ? We now that the class is `InlinePrint` print here. It cannot be a child class as it is final.

Reply:
indeed, just a copy/paste, fixed.